workbench/src/main/setupAddRemovePlugin.js

```diff
@@ -52,32 +52,56 @@ function spawnWithLogging(cmd, args, options) {
 export function setupAddPlugin() {
   ipcMain.handle(
     ipcMainChannels.ADD_PLUGIN,
-    async (e, pluginURL) => {
+    async (e, url, revision, path) => {
       try {
-        logger.info('adding plugin at', pluginURL);
-        const micromamba = settingsStore.get('micromamba');
-        // Create a temporary directory and check out the plugin's pyproject.toml
-        const tmpPluginDir = fs.mkdtempSync(upath.join(tmpdir(), 'natcap-invest-'));
-        await spawnWithLogging(
-          'git',
-          ['clone', '--depth', '1', '--no-checkout', pluginURL, tmpPluginDir]
-        );
-        await spawnWithLogging(
-          'git',
-          ['checkout', 'HEAD', 'pyproject.toml'],
-          { cwd: tmpPluginDir }
-        );
-        // Read in the plugin's pyproject.toml, then delete it
-        const pyprojectTOML = toml.parse(fs.readFileSync(
-          upath.join(tmpPluginDir, 'pyproject.toml')
-        ).toString());
-        fs.rmSync(tmpPluginDir, { recursive: true, force: true });
+        let pyprojectTOML;
+        let installString;
+        if (url) { // install from git URL
+          if (revision) {
+            installString = `git+${url}@${revision}`;
+            logger.info(`adding plugin from ${installString}`);
+          } else {
+            installString = `git+${url}`;
+            logger.info(`adding plugin from ${installString} at default branch`);
+          }
+
+          // Create a temporary directory and check out the plugin's pyproject.toml,
+          // without downloading any extra files or git history
+          const tmpPluginDir = fs.mkdtempSync(upath.join(tmpdir(), 'natcap-invest-'));
```

Comment:
Is the trailing dash in `natcap-invest-` something common for a temp directory? Just curious!

Reply:
It's a prefix for the temp dir name, so the actual name will be something like `natcap-invest-xkdfdjlksdad`